### PR TITLE
Update for .NET Core support

### PR DESCRIPTION
**Pull request by eogas**
Most of the changes are migrations from the project.json and xproj to a new csproj, and a few changes for updates in the APIs.  I've added comments where I felt some explanation was necessary.  I did some light testing just to make sure the app runs and a vsix can be uploaded and served correctly.

On an unrelated note, would it be possible to add a LICENSE to the project? I was interested in hosting an internal instance for some extensions I've written for work, but I wouldn't want to do so without a license on the code.

### CODE REVIEW

File: src/ExtensionGallery/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (eogas) [inline src/ExtensionGallery/.vscode/launch.json:1]: I accidentally committed my .vscode directory without thinking.  These can definitely be excluded if desired.

File: src/ExtensionGallery/Controllers/ApiController.cs [hunk 2]
```diff
@@ -46,12 +49,17 @@ public object Get(string id)
...
 
 		[HttpPost]
-		public async Task<IActionResult> Upload([FromQuery]string repo, string issuetracker)
+		public async Task<IActionResult> Upload()
 		{
+			var context = _httpContextAccessor.HttpContext;
+
+			Stream bodyStream = context.Request.Body;
+			string repo = context.Request.Query["repo"];
+			string issueTracker = context.Request.Query["issuetracker"];
```
Comment (eogas) [inline src/ExtensionGallery/Controllers/ApiController.cs:58]: For some reason, .NET Core seems to ignore the request body if the querystring is being auto-parsed into arguments.  I just pulled them out manually as a workaround.

File: src/ExtensionGallery/ExtensionGallery.xproj [hunk 3]
```diff
@@ -1,30 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup>
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
-    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
-  </PropertyGroup>
-  <PropertyGroup Label="Globals">
-    <ProjectGuid>5092448c-b511-4077-b31c-00685e56fb2d</ProjectGuid>
-    <RootNamespace>ExtensionGallery</RootNamespace>
-    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
```
Comment (eogas) [inline src/ExtensionGallery/ExtensionGallery.xproj:10]: I'm not sure what this artifacts directory is all about.  I haven't accounted for this in the migration.

File: src/ExtensionGallery/Startup.cs [hunk 4]
```diff
@@ -1,109 +1,63 @@
...
 namespace ExtensionGallery
 {
 	public class Startup
 	{
-		public Startup(IHostingEnvironment env, IApplicationEnvironment appEnv)
+		public Startup(IHostingEnvironment env)
 		{
             // Setup configuration sources.
-            Configuration = new ConfigurationBuilder(appEnv.ApplicationBasePath)
-				.AddJsonFile("config.json")
```
Comment (eogas) [inline src/ExtensionGallery/Startup.cs:22]: It looked like the config wasn't being used anyway, so I just used the default config setup that the tooling generates.

File: src/ExtensionGallery/Startup.cs [hunk 5]
```diff
@@ -1,109 +1,63 @@
...
-			services.AddMvc();
+			services.AddMvc()
+                .AddJsonOptions(options => options.SerializerSettings.ContractResolver = new DefaultContractResolver());
 
-			// Uncomment the following line to add Web API servcies which makes it easier to port Web API 2 controllers.
-			// You need to add Microsoft.AspNet.Mvc.WebApiCompatShim package to project.json
-			// services.AddWebApiConventions();
-
-		}
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
```
Comment (eogas) [inline src/ExtensionGallery/Startup.cs:39]: The IHttpContextAccessor service is not registered by default, per https://github.com/aspnet/Hosting/issues/793

File: src/ExtensionGallery/Startup.cs [hunk 6]
```diff
@@ -1,109 +1,63 @@
...
-			//    .AddSqlServer()
-			//    .AddDbContext<ApplicationDbContext>();
-
-			//// Add Identity services to the services container.
-			//services.AddDefaultIdentity<ApplicationDbContext, ApplicationUser, IdentityRole>(Configuration);
-
 			// Add MVC services to the services container.
-			services.AddMvc();
+			services.AddMvc()
+                .AddJsonOptions(options => options.SerializerSettings.ContractResolver = new DefaultContractResolver());
```
Comment (eogas) [inline src/ExtensionGallery/Startup.cs:37]: MVC now serializes JSON with camel case names by default, per https://github.com/aspnet/Announcements/issues/194

Leaving camelcase as default causes a bunch of stuff to break in the angular code.  The better fix is probably to rewrite the JS to assume camelcase, but that's a hassle, so I just used this workaround for now.

**Comment by madskristensen** [comment]
I appreciate the effort, but my plan is to completely rewrite the site, not just update it. Sorry about all your hard work on this, but it doesn't align with my plans.

Thanks

**Comment by eogas** [comment]
No worries, it wasn't that much work.  Thanks for the quick response.

**Pull request closed without merging**